src/Components/App/App.tsx

```diff
@@ -43,46 +47,72 @@ class App extends Component<{}, IAppState> {
       })
   }
 
+  findPostsWithSearchTerm = (searchTerm: string) => {
+      searchTerm = searchTerm.toLowerCase()
+    return this.setState({ foundPosts: this.state.allPosts.filter(post => {
+           return post.content.toLowerCase().includes(searchTerm) || post.title.toLowerCase().includes(searchTerm) || post.author.toLowerCase().includes(searchTerm)
+        })})
+  }
+
+  resetFoundPosts = () => {
+    this.setState({ foundPosts: [] })
+  }
+
   renderComponent = () => {
-    if (this.state.allPosts.length > 0) {
+    if (this.state.foundPosts.length > 0) {
       return (
-        <section className='main-page'>
-          <Searchbar />
-          <AllPosts allPosts={this.state.allPosts} />
-          <Nav />
-        </section>
+        <>
+            <Search findPostsWithSearchTerm={this.findPostsWithSearchTerm} />
+            <AllPosts allPosts={this.state.foundPosts} />
+            <Nav resetFoundPosts={this.resetFoundPosts}/>
+        </>
       )
-    } else if (this.state.error) {
+    } else if (this.state.allPosts.length > 0) {
       return (
-        <h2>{this.state.error}</h2>
-      )
-    } else {
-      return (
-        <Loading />
+        <>
+          <Search findPostsWithSearchTerm={this.findPostsWithSearchTerm} />
+          <AllPosts allPosts={this.state.allPosts} />
+          <Nav resetFoundPosts={this.resetFoundPosts}/>
+        </>
       )
     }
   }
 
+  // findPost = () => {
+  //   const currentPost = this.state.allPosts.find(post => {
+  //     return post.pid ===
+  //   })
+  // }
+
   render() {
     return (
       <main className='app'>
-        <Switch>
-          <Route
-            exact path="/"
-            render={this.renderComponent}
-          />
-          <Route
-            exact path="/new-post"
-            render={() => {
-              return <NewPostForm addNewPost={this.addNewPost} />
-            }
-            }
-          />
-          <Route
-            exact path='/:pid'
-            render={({ match }: RouteComponentProps<TParams>) => <ExpandedPost match={match.params.pid}/>}
-          />
-        </Switch>
+          {!!this.state.error &&
```

Comment:
What does the "!!" mean here?

Reply:
It's called a double bang operator, and it's basically checking to see if a non-boolean value is truthy. I've only used it a handful of times before, and Leta used it on one of our class projects (one of the Ideabox apps, I believe).  Here's an article:  https://dev.to/sanchithasr/what-is-the-double-bang-operator-in-javascript-4i3h